### PR TITLE
feat(dashboard): Make FilterBar width resizable

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY

The current native filterbar section is fixed size.
Since the filter is not wide enough for moderately wide data to be seen, some filter value with the long overlapped prefix string such like the following case truncated with the repeated strings.

Screen Shot 2022-07-14 at 1 46 21 PM

There is an HTML tooltip that appear on mouseover but it's slow and bad user experience.
The dropdown needs to be widened over the sidebar but the dropdown also truncated under the sidebar to fix the  https://github.com/apache/superset/pull/19399 

Screen Shot 2022-07-19 at 2 14 41 PM

Therefore, this commit makes the sidebar resizable to be widened itself.
This commit adds the Resizable panel for native filterbar to adjust the width wider to help this inconvenient case.
It also stores the adjusted width for individual dashboard in localStorage to memorize the state.

Addition to the feature, this commit also migrates DashboardBuilder test code to testing-library, jest and typescript to enable emotion test helpers.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Before:

https://user-images.githubusercontent.com/1392866/179843448-425ad37d-9a6d-409e-abdc-b78cfa2f4412.mov

After:

https://user-images.githubusercontent.com/1392866/179843423-4effc912-42d3-4811-8d2a-6aeacd7b0d79.mov

### TESTING IN

STRUCTIONS

- Go to dashboard and then open the native filter menu
- hover on the right edge of the menu
- drag to adjust the width

### ADDITIONAL INFORMATION
- [x] Has associated issue:
- [x] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/constants.ts [hunk 1]
```diff
@@ -37,6 +37,7 @@ export const PLACEHOLDER_DATASOURCE: Datasource = {
 export const MAIN_HEADER_HEIGHT = 53;
 export const CLOSED_FILTER_BAR_WIDTH = 32;
 export const OPEN_FILTER_BAR_WIDTH = 260;
```
Comment (ktmud) [inline superset-frontend/src/dashboard/constants.ts:39]: Maybe rename this to `OPEN_FILTER_BAR_DEFAULT_WIDTH`?

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx [hunk 2]
```diff
@@ -85,6 +86,7 @@ const ActionButtonsContainer = styled.div`
 `;
 
 export const ActionButtons = ({
+  width,
```
Comment (ktmud) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx:89]:
```suggestion
  width = OPEN_FILTER_BAR_WIDTH
```

Nit: let's assign a default value as soon as possible

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx [hunk 3]
```diff
@@ -103,7 +105,7 @@ export const ActionButtons = ({
   );
 
   return (
-    <ActionButtonsContainer>
+    <ActionButtonsContainer data-test="filterbar-action-buttons" width={width || OPEN_FILTER_BAR_WIDTH}>
```
Comment (ktmud) [inline superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx]:
```suggestion
    <ActionButtonsContainer data-test="filterbar-action-buttons" width={width}>
```

No need for `||` is default value is assigned.

**Review by ktmud — COMMENTED** [review]
LGTM if CI passes. Thanks for the comprehensive tests.

**Review by ktmud — APPROVED** [review]

File: superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx (hunk 2 above)
Comment (justinpark) [reply superset-frontend/src/dashboard/components/nativeFilters/FilterBar/ActionButtons/index.tsx:89]: sounds good

File: superset-frontend/src/dashboard/constants.ts (hunk 1 above)
Comment (justinpark) [reply superset-frontend/src/dashboard/constants.ts:39]: I kept this variable because some custom override might be impacted.

**Pull request merged**